examples/hello_world_util.h

```diff
@@ -48,18 +47,68 @@
 static ucs_memory_type_t test_mem_type = UCS_MEMORY_TYPE_HOST;
 
 
-#define CUDA_FUNC(_func)                                   \
+#define CUDADRV_FUNC(_func)                                \
     do {                                                   \
-        cudaError_t _result = (_func);                     \
-        if (cudaSuccess != _result) {                      \
+        CUresult _result = (_func);                        \
+        const char *_cu_err_str;                           \
+        if (CUDA_SUCCESS != _result) {                     \
+            cuGetErrorString(_result, &_cu_err_str);       \
             fprintf(stderr, "%s failed: %s\n",             \
-                    #_func, cudaGetErrorString(_result));  \
+                    #_func, _cu_err_str);                  \
         }                                                  \
     } while(0)
 
 
 void print_common_help(void);
 
+#ifdef HAVE_CUDA
+CUcontext cuda_ctx;
+int cuda_ctx_set = 0;
+
+int init_cuda_context()
```

Comment:
Does it mean that user should always initialize the device and create the context manually to use UCX with CUDA?
Otherwise all driver API calls will return `CUDA_ERROR_NOT_INITIALIZED` or `CUDA_ERROR_INVALID_CONTEXT`.

Reply:
Yes

Reply:
Sorry, the comment is not relevant.
Noticed that `cuInit` is called inside static initialization in `cuda_md.c`.